src/components/CheckIn/AddPieceModal.js

```diff
@@ -49,26 +54,75 @@ const footer = (close, save, checkIn) => (
 
 class AddPieceModal extends Component {
   static propTypes = {
+    change: PropTypes.func.isRequired,
     checkIn: PropTypes.func.isRequired,
     close: PropTypes.func.isRequired,
+    createInventoryValues: PropTypes.object.isRequired,
+    dispatch: PropTypes.func.isRequired,
     handleSubmit: PropTypes.func.isRequired,
     locations: PropTypes.arrayOf(PropTypes.shape({
       label: PropTypes.string.isRequired,
       value: PropTypes.string.isRequired,
     })),
     showPieceFormatField: PropTypes.bool,
-    store: PropTypes.object.isRequired,
+    stripes: PropTypes.object.isRequired,
+  }
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      isAddItemButtonDisabled: true,
+    };
   }
 
-  checkIn = () => this.props.checkIn(getFormValues('AddPieceModalForm')(this.props.store.getState()));
+  getValues = () => {
+    const { stripes: { store } } = this.props;
+
+    return getFormValues(ADD_PIECE_MODAL_FORM)(store.getState());
+  }
+
+  checkIn = () => this.props.checkIn(this.getValues());
+
+  onChangeSelect = (_, value, prevValue, fieldName) => {
+    const { dispatch, change } = this.props;
+
+    dispatch(change(fieldName, value));
+    this.isAddItemButtonDisabled();
+  }
+
+  isAddItemButtonDisabled = () => {
```

Comment:
for me function with such name should return boolean

Reply:
Removed this function. Now I check `isAddItemButtonDisabled` in  render